<html>
<head>
<title>crux comet</title>
<link href="./crux.css" rel="styleSheet" type="text/css">
</head>
<body>
<blockquote>

<H1 ALIGN=CENTER>crux comet</H1>

<h2>Usage:</h2>

<blockquote>
<code>crux comet [options]
&lt;spectra&gt; &lt;protein&nbsp;input&gt;
</code>
</blockquote>

<h2>Description:</h2>

<blockquote>
<p>
This command searches a protein database with a set of spectra,
assigning peptide sequences to the observed spectra.  This command was
developed by Jimmy Eng at the University of Washington Proteomics
Resource.</p>

<p>
Although its history goes back two decades, the Comet search engine
was first made publicly available in August 2012 on SourceForge under
the Apache License, version 2.0. Comet is multithreaded and supports
multiple input and output formats.  The current Comet version in
Crux is FIXME.</p>

<blockquote>
"Comet: an open source tandem mass spectrometry sequence database
search tool."  Eng JK, Jahan TA, Hoopmann MR.
<i>Proteomics</i>. 2012 Nov 12. doi: 10.1002/pmic.201200439
</blockquote>

</blockquote>

<p>
<h2>Input:</h2>

<ul>

<li>
&lt;spectra&gt; &ndash; The name of the file from which to parse the
  spectra.  File formats are supported by 
  <a href="http://proteowizard.sourceforge.net/">proteowizard</a>, 
  with the exception of the vendor's formats.</li>

<LI>
&lt;database_name&gt; &ndash; The name of the file in fasta format
  from which to retrieve proteins and peptides.  The database can
  contain amino acid sequences or nucleic acid sequences.  If the
  sequences are nucleic acid sequences, then you must instruct Comet
  to translate these to amino acid sequences by setting
  <code>nucleotide_reading_frame</code> to a value between 1 and
  9.</li>

</UL>

<h2>Output:</h2>

<p>
The program writes files to the folder <code>crux-output</code> by
default.  The name of the output folder can be set by the user using
the <code>--output-dir</code> option.  The following files will be
created:</p>
<ul>

<li>
<b>comet.params.txt</b>: a file containing the name and value of all
parameters/options for the current operation.  Not all parameters in
the file may have been used in the operation.  The resulting file can
be used with the <code>--parameter-file</code> option for other crux
programs.
</li>

<li>
<b>comet.target.pep.xml</b>: a file in pep.xml format containing the
PSMs.  See 
<a href="http://tools.proteomecenter.org/wiki/index.php?title=Formats:pepXML">
pep xml file format</a> for further reference. This file can be used as input 
to some of the tools in the 
<a href="http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP">
Transproteomic Pipeline</a>.</li>

<li>
<b>comet.log.txt</b>: a log file containing a copy of all messages
that were printed to standard error.</li>

</ul>

<h2>Options:</h2>

<p>
Because Comet was developed externally to Crux, the names of various
command line parameters are somewhat different.  For example, Comet
uses underscores ("_") within parameter names, whereas other Crux
commands use hyphens ("-").  The <code>crux comet</code>
command will accept parameters using either convention.</p>

<ul>

<h3>Database</h3>
<ul>
<li><a href="comet-options/decoy_search.html">decoy_search</a></li>
</ul>

<h3>CPU threads</h3>
<ul>
<li><a href="comet-options/num_threads.html">num_threads</a></li>
</ul>

<h3>Masses</h3>
<ul>
<li><a href="comet-options/peptide_mass_tolerance.html">peptide_mass_tolerance</a></li>
<li><a href="comet-options/peptide_mass_units.html">peptide_mass_units</a></li>
<li><a href="comet-options/mass_type_parent.html">mass_type_parent</a></li>
<li><a href="comet-options/mass_type_fragment.html">mass_type_fragment</a></li>
<li><a href="comet-options/precursor_tolerance_type.html">precursor_tolerance_type</a></li>
<li><a href="comet-options/isotope_error.html">isotope_error</a></li>
</ul>

<h3>Search enzyme</h3>
<ul>
<li><a href="comet-options/search_enzyme_number.html">search_enzyme_number</a></li>
<li><a href="comet-options/num_enzyme_termini.html">num_enzyme_termini</a></li>
<li><a href="comet-options/allowed_missed_cleavage.html">allowed_missed_cleavage</a></li>
</ul>

<h3>Fragment ions</h3>
<ul>
<li><a href="comet-options/fragment_bin_tol.html">fragment_bin_tol</a></li>
<li><a href="comet-options/fragment_bin_offset.html">fragment_bin_offset</a></li>
<li><a href="comet-options/theoretical_fragment_ions.html">theoretical_fragment_ions</a></li>
<li><a href="comet-options/use_A_ions.html">use_A_ions</a></li>
<li><a href="comet-options/use_B_ions.html">use_B_ions</a></li>
<li><a href="comet-options/use_C_ions.html">use_C_ions</a></li>
<li><a href="comet-options/use_X_ions.html">use_X_ions</a></li>
<li><a href="comet-options/use_Y_ions.html">use_Y_ions</a></li>
<li><a href="comet-options/use_Z_ions.html">use_Z_ions</a></li>
<li><a href="comet-options/use_NL_ions.html">use_NL_ions</a></li>
</ul>

<h3>Output</h3>
<ul>
<li>
<code>--fileroot &lt;string&gt;</code> &ndash; 
The <code>fileroot</code> string will be added as a prefix to all
output file names. Default = none.</li>

<li>
<code>--output-dir &lt;filename&gt;</code> &ndash; 
The name of the directory where output files will be created.
Default = crux-output.</li>

<li>
<code>--overwrite &lt;T|F&gt;</code> &ndash;
Replace existing files if true (T) or fail when trying to overwrite a
file if false (F). Default =  F.</li>

<li>
<code>--decoy-prefix &lt;string&gt;</code> &ndash; Specifies the
prefix of the protein names that indicates a decoy. Default = rand_.
</li>

<li><a href="comet-options/output_sqtfile.html">output_sqtfile</a></li>
<li><a href="comet-options/output_pepxmlfile.html">output_pepxmlfile</a></li>
<li><a href="comet-options/output_outfiles.html">output_outfiles</a></li>
<li><a href="comet-options/print_expect_score.html">print_expect_score</a></li>
<li><a href="comet-options/num_output_lines.html">num_output_lines</a></li>
<li><a href="comet-options/show_fragment_ions.html">show_fragment_ions</a></li>
<li><a href="comet-options/sample_enzyme_number.html">sample_enzyme_number</a></li>
</ul>
               
<h3>mzXML/mzML parameters</h3>
<ul>
<li><a href="comet-options/scan_range.html">scan_range</a></li>
<li><a href="comet-options/precursor_charge.html">precursor_charge</a></li>
<li><a href="comet-options/ms_level.html">ms_level</a></li>
<li><a href="comet-options/activation_method.html">activation_method</a></li>
</ul>

<h3>Miscellaneous parameters</h3>
<ul>
<LI><code>--parameter-file &lt;filename&gt;</code> &ndash; 
A file containing command-line or additional parameters.  See
the <a href="parameter-file.html">parameter documentation</a> page
for details.  Default = no parameter file.</li>

<LI>
<code>--verbosity &lt;0-100&gt;</code> &ndash; 
Specify the verbosity of the current processes. Each level prints the
following messages, including all those at lower verbosity levels:
0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on
the progress of execution, 40-more progress information, 50-debug
info, 60-detailed debug info. Default = 30.</li>

<li><a href="comet-options/digest_mass_range.html">digest_mass_range</a></li>
<li><a href="comet-options/num_results.html">num_results</a></li>
<li><a href="comet-options/skip_researching.html">skip_researching</a></li>
<li><a href="comet-options/max_fragment_charge.html">max_fragment_charge</a></li>
<li><a href="comet-options/max_precursor_charge.html">max_precursor_charge</a></li>
<li><a href="comet-options/nucleotide_reading_frame.html">nucleotide_reading_frame</a></li>
<li><a href="comet-options/clip_nterm_methionine.html">clip_nterm_methionine</a></li>
</ul>

<h3>Spectral processing</h3>
<ul>
<li><a href="comet-options/minimum_peaks.html">minimum_peaks</a></li>
<li><a href="comet-options/minimum_intensity.html">minimum_intensity</a></li>
<li><a href="comet-options/remove_precursor_peak.html">remove_precursor_peak</a></li>
<li><a href="comet-options/remove_precursor_tolerance.html">remove_precursor_tolerance</a></li>
</ul>
               
<h3>Variable modifications</h3>
<ul>
<li><a href="comet-options/variable_mod1.html">variable_mod1</a></li>
<li><a href="comet-options/variable_mod2.html">variable_mod2</a></li>
<li><a href="comet-options/variable_mod3.html">variable_mod3</a></li>
<li><a href="comet-options/variable_mod4.html">variable_mod4</a></li>
<li><a href="comet-options/variable_mod5.html">variable_mod5</a></li>
<li><a href="comet-options/variable_mod6.html">variable_mod6</a></li>
<li><a href="comet-options/max_variable_mods_in_peptide.html">max_variable_mods_in_peptide</a></li>
<li><a href="comet-options/variable_C_terminus.html">variable_C_terminus</a></li>
<li><a href="comet-options/variable_N_terminus.html">variable_N_terminus</a></li>
<li><a href="comet-options/variable_C_terminus_distance.html">variable_C_terminus_distance</a></li>
<li><a href="comet-options/variable_N_terminus_distance.html">variable_N_terminus_distance</a></li>
</ul>

<h3>Static modifications</h3>
<ul>
<li><a href="comet-options/add_Cterm_peptide.html">add_Cterm_peptide</a></li>
<li><a href="comet-options/add_Nterm_peptide.html">add_Nterm_peptide</a></li>
<li><a href="comet-options/add_Cterm_protein.html">add_Cterm_protein</a></li>
<li><a href="comet-options/add_Nterm_protein.html">add_Nterm_protein</a></li>
<li><a href="comet-options/add_G_glycine.html">add_G_glycine</a></li>
<li><a href="comet-options/add_A_alanine.html">add_A_alanine</a></li>
<li><a href="comet-options/add_S_serine.html">add_S_serine</a></li>
<li><a href="comet-options/add_P_proline.html">add_P_proline</a></li>
<li><a href="comet-options/add_V_valine.html">add_V_valine</a></li>
<li><a href="comet-options/add_T_threonine.html">add_T_threonine</a></li>
<li><a href="comet-options/add_C_cysteine.html">add_C_cysteine</a></li>
<li><a href="comet-options/add_L_leucine.html">add_L_leucine</a></li>
<li><a href="comet-options/add_I_isoleucine.html">add_I_isoleucine</a></li>
<li><a href="comet-options/add_N_asparagine.html">add_N_asparagine</a></li>
<li><a href="comet-options/add_D_aspartic_acid.html">add_D_aspartic_acid</a></li>
<li><a href="comet-options/add_Q_glutamine.html">add_Q_glutamine</a></li>
<li><a href="comet-options/add_K_lysine.html">add_K_lysine</a></li>
<li><a href="comet-options/add_E_glutamic_acid.html">add_E_glutamic_acid</a></li>
<li><a href="comet-options/add_M_methionine.html">add_M_methionine</a></li>
<li><a href="comet-options/add_O_ornithine.html">add_O_ornithine</a></li>
<li><a href="comet-options/add_H_histidine.html">add_H_histidine</a></li>
<li><a href="comet-options/add_F_phenylalanine.html">add_F_phenylalanine</a></li>
<li><a href="comet-options/add_R_arginine.html">add_R_arginine</a></li>
<li><a href="comet-options/add_Y_tyrosine.html">add_Y_tyrosine</a></li>
<li><a href="comet-options/add_W_tryptophan.html">add_W_tryptophan</a></li>
<li><a href="comet-options/add_B_user_amino_acid.html">add_B_user_amino_acid</a></li>
<li><a href="comet-options/add_J_user_amino_acid.html">add_J_user_amino_acid</a></li>
<li><a href="comet-options/add_U_user_amino_acid.html">add_U_user_amino_acid</a></li>
<li><a href="comet-options/add_X_user_amino_acid.html">add_X_user_amino_acid</a></li>
<li><a href="comet-options/add_Z_user_amino_acid.html">add_Z_user_amino_acid</a></li>
</ul>


</UL>


</ul>

<note>

 

<p>
To do:</p>

<ul>
<li>
Resolve MSToolkit issues (Comet has its own copy, we need
MSToolkit for harklor/bullseye.)  Also, latest version of comet has a version of MSToolkit that doesn't compile on MAC.
</li>

<li>Figure out how to either 1) tie duplicate parameters together or 2) make crux use comet parameters so when when crux writes out
the parameter file, this can instantly be used by calling comet.</li>

<li>Handle specifying enzymes.  Comet does this with the <code>search_enzyme_number</code> and the following lines at the end of the parameter file:<br>
<code><br>
#<br>
# COMET_ENZYME_INFO _must_ be at the end of this parameters file<br>
#<br>
[COMET_ENZYME_INFO]<br>
0.  No_enzyme              0      -           -<br>
1.  Trypsin                1      KR          P<br>
2.  Trypsin/P              1      KR          -<br>
3.  Lys_C                  1      K           P<br>
4.  Lys_N                  0      K           -<br>
5.  Arg_C                  1      R           P<br>
6.  Asp_N                  0      D           -<br>
7.  CNBr                   1      M           -<br>
8.  Glu_C                  1      DE          P<br>
9.  PepsinA                1      FL          P<br>
10. Chymotrypsin           1      FWYL        P<br>
</code>

The <code>search_enzyme_number</code> keys into this list of enzymes.  This can be modified to support other enzymes.
If we go with the mixed crux/comet parameters, then this will have to be put at the end of the parameter file and the rest of the crux programs
will have to be modified to ignore this list.
</li>

</ul>

<p>
Changes we are proposing on the Comet end of things:</p>
<ul>
<li>Have an option to compile Comet as a library in the with the main() replaced with comet_main().</li>
<li>Make sure comet ignores parameters that are not part of comet (i.e. part of crux).</li>
<li>Allow an option that can specify the full paths of the results files.</li>
<li>Map the modifications to actual mass changes instead of using special characters.</li>
</ul>

<p>
Questions:
</p>

<ul>
<li>When a decoy search is performed, how are the decoys indicated in
  the output?</li>


</ul>
<!--
Delete sometime in the future

<p>Argument parameter mappings</p>
<br>
    Comet usage:  ../comet.exe [options] <input_files>[:range]<br>
    The optional [:range] parameter specifies a scan range to search<br>
    <table border="1">
      <tr>
        <td>Comet parameter</td>

        <td>Parameter File/Command Line</td>
        <td>Parameter Description</td>
        <td>Crux Parameter</td>
        <td>Exists in Crux</td>
        <td>Priority</td>
        <td>Notes</td>
      </tr>
      <tr>
        <td>-p</td>
        <td>Command Line</td>

        <td>print out comet.params.new</td>
        <td></td>
        <td>Do not implement</td>
      </tr>
      <tr>
        <td>-P params</td>
        <td>Command Line</td>
        <td>specify an alternate parameter file</td>
        <td>Override this with our generated parameter file</td>
        <td>N/A</td>
        <td>1</td>
      </tr>
      <tr>
        <td>-N name</td>
        <td>Command Line</td>
        <td>specifiy an alterate output base name</td>
        <td>Override this with crux-output</td>
        <td>Yes</td>
        <td>1</td>
      </tr>
      <tr>
        <td>-D dbase</td>
        <td>Command Line</td>
        <td>specify a sequence database, overrideing entry in parameters file</td>
        <td><code>protein database</code></td>
        <td>Yes</td>
        <td>Do not implement</td>
      </tr>

      <tr>
        <td>[:range]</td>
        <td>Command Line</td>
        <td>Specify range of scans to search</td>
        <td><code>--scan-number</code</td>
        <td>Yes</td>
        <td>7</td>
      </tr>
      <tr>

        <td><code>database_name</code></td>
        <td>Parameter File</td>
        <td>Fasta file of database</td>
        <td><code>protein database</code></td>
        <td>Yes</td>
        <td>1</td>
      </tr>
      <tr>
        <td><code>decoy_search</code></td>
        <td>Parameter File</td>
        <td>0= no, 1=concatenated, 2=separate search</td>
        <td><code>--decoy-location</code>,<code>--num-decoys-per-target</code></td>
        <td>Yes</td>
      </tr>
      <tr>
        <td><code>num_threads</code></td>
        <td>Parameter File</td>
        <td>0=poll CPU to set threads; </td>
        <td><code>--num-threads</code></td>
        <td>No</td>
        <td>2</td>
      </tr>
      <tr>
        <td colspan="7"><b>masses</b></td>
      </tr>
      <tr>
        <td><code>peptide_mass_tolerance</code></td>
        <td>Parameter File</td>
        <td>Mass tolerance to use when selecting candidate peptides</td>
        <td><code>--precursor-window</td>
        <td>Yes</td>
        <td>2</td>    
      </tr>
      <tr>
        <td><code>peptide_mass_units</code></td>
        <td>Parameter File</td>
        <td>Units to use when selecting peptides, 0=amu, 1=mmu, 2=ppm</td>
        <td><code>--precursor-window-type</code></td>
        <td>Yes</td>
        <td>2</td>
      </tr>
      <tr>
        <td><code>mass_type_parent</code></td>
        <td>Parameter File</td>
        <td>average or monoisotopic mass for calculating peptide mass 0=average mass, 1=monoisotopic mass</td>
        <td><code>--isotopic-mass</code>
        <td>Yes</td>
        <td>2</td>
      </tr>

      <tr>
        <td><code>mass_type_fragment</code></td>
        <td>Parameter File</td>
        <td>average or monoisotopic mass for calculating fragment masses</td>
        <td><code>--fragment-mass</code>
        <td>Yes</td>
        <td>2</td>
      </tr>
      <tr>

        <td><code>precursor_tolerance_type</code></td>
        <td>Parameter File</td>
        <td>0=MH+, 1=precursor m/z</td>
        <td>???</td>
      </tr>
      <tr>
        <td><code>isotope_error</code></td>
        <td>Parameter File</td>
        <td>Search in small windows that are +/- 1 Neutron Mass away from precursor mass?</td>
        <td><code>--isotope-windows</td>
        <td>No</td>
        <td>7</td>
      </tr>
      <tr>
        <td colspan="7"><b>search enzyme</b></td>
      </tr>
      <tr>
        <td>search_enzyme_number</td>
        <td>Parameter File</td>
        <td>Enzyme to use when digesting peptides</td>
        <td><code>--enzyme</code></td>
        <td>Yes</td>
        <td>4</td>
      </tr>
      <tr>
        <td>num_enzyme_termini</td>
        <td>Parameter File</td>
        <td>selecting semi-digestion, full digestion, and others valid values are 1 (semi-digested), 2 (fully digested, default), 8 N-term, 9 C-term</td>
        <td><code>digestion</code></td>
        <td>Yes</td>
        <td><code>digestion</code> only supports partial and full</td>
        <td>7</td>
      </tr>
      <tr>
        <td>allowed_missed_cleavage</td>
        <td>Parameter File</td>
        <td>Number of missed cleavages allowed (max 5)</td>
        <td><code>--missed-cleaveages</code></td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>variable_modX (X=1-6)</td>
        <td>Parameter File</td>
        <td>Variable modification support</td>
        <td>Supported using mod</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>max_variable_mods_in_peptide</td>
        <td>Parameter File</td>
        <td>Maximum number of mods to consider in peptide</td>
        <td><code>--max-mods</code></td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>fragment_bin_tol</td>
        <td>Parameter File</td>
        <td>bin width to use on fragment ions</td>
        <td>mz-bin-width</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>fragment_bin_offset</td>
        <td>Parameter File</td>
        <td>offset to start the binning</td>
        <td>mz-bin-offset</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>theoretical_fragment_ions</td>
        <td>Parameter File</td>
        <td>??? 0=default peak shape, 1=M peak only</td>
        <td></td>
        <td>No</td>
      </tr>
      <tr>
        <td>use_A_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use a-ions</td>
        <td><code>use-a-ions</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>use_B_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use b-ions</td>
        <td><code>use-b-ions</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>use_C_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use c-ions</td>
        <td><code>use-c-ions</code></td>
        <td>No</td>
      </tr>        
      <tr>
        <td>use_X_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use x-ions</td>
        <td><code>use-x-ions</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>use_Y_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use y-ions</td>
        <td><code>use-y-ions</code></td>
        <td>No</td>
      </tr>

      <tr>
        <td>use_Z_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use z-ions</td>
        <td><code>use-z-ions</code></td>
        <td>No</td>
      </tr>
       <tr>
        <td>use_NL_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use NH3/H2O neutral loss peaks</td>
        <td><code>use-nl-ions</code></td>
        <td>No</td>
      </tr>     
      <tr>
        <td>output_sqtstream</td>
        <td>Parameter File</td>
        <td>0=no, 1=yes  write sqt to standard output</td>
        <td>Not supported</td>
        <td>No</td>
      </tr>
      <tr>
        <td>output_sqtfile</td>
        <td>Parameter File</td>
        <td>0=no, 1=yes  write sqt file</td>
        <td>Always true</td>
        <td>No</td>
      </tr>
      <tr>
        <td>output_pepxmlfile</td>
        <td>Parameter File</td>
        <td>0=no, 1=yes  write pep.xml file</td>
        <td>Always true</td>
        <td>No</td>
      </tr>
      <tr>
        <td>output_outfiles</td>
        <td>Parameter File</td>
        <td>0=no, 1=yes  write .out files</td>
        <td>Always false</td>
        <td>No</td>
      </tr>
      <tr>
        <td>print_expect_score</td>
        <td>Parameter File</td>
        <td>to replace Sp with expect in out & sqt</td>
        <td><code>print-expect-score</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>num_output_lines</td><td>Parameter File</td>
        <td>num peptide results to show</td>
        <td><code>top-match</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>show_fragment_ions</td>
        <td>Parameter File</td>
        <td> 0=no, 1=yes for out files only</td>
        <td>Always false</td>
        <td>No</td>
      </tr>
      <tr>
        <td>sample_enzyme_number</td>
        <td>Parameter File</td>
        <td>Sample enzyme which is possibly different than the one applied to the search.<br>
          Used to calculate NTT & NMC in pepXML output (default=1 for trypsin).</td>
        <td><code>sample-enzymze</code></td>
        <td>No</td>
      </tr>
      <tr>
         <td colspan="7"><B>mzXML parameters</B></td>
      </tr>
      <tr>
        <td>scan_range</td><td>Parameter File</td>
        <td>start and scan scan range to search; 0 as 1st entry ignores parameter</td>
        <td><code>scan-number</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>precursor_charge</td><td>Parameter File</td>
        <td>precursor charge range to analyze; does not override mzXML charge; 0 as 1st entry ignores parameter</td>
        <td><code>spectrum-charge</code></td>
        <td>Yes, we need to modify it to accept a range of charges</td>
      </tr>
      <tr>
         <td>ms_level</td><td>Parameter File</td>
         <td>MS level to analyze, valid are levels 2 (default) or 3</td>
         <td><code>ms-level</code></td>
         <td>No</td>
      </tr>
      <tr>
        <td>activation_method</td>
        <td>Parameter File</td>
        <td>activation method; used if activation method set; allowed ALL, CID, ECD, ETD, PQD, HCD, IRMPD</td>
        <td><code>activation-method</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td colspan="7"><B>misc parameters</B></td>
      </tr>
      <tr>
        <td>digest_mass_range</td>
        <td>Parameter File</td>
        <td>MH+ peptide mass range to analyze</td>
        <td><code>min-mass,max-mass</code></td>
        <td>No</td> 
      </tr>
      <tr>
        <td>num_results</td>
        <td>Parameter File</td>
        <td>number of search hits to store internally</td>
        <td>Default it to 50</td>
        <td>No</td>
      </tr>
      <tr>
        <td>skip_researching</td>
        <td>Parameter File</td>
        <td>for '.out' file output only, 0=search everything again (default), 1=don't search if .out exists</td>
        <td>Default to 0</td>
        <td>No</td>
      </tr>
      <tr>
        <td>max_fragment_charge</td>
        <td>Parameter File</td>
        <td>set maximum fragment charge state to analyze (allowed max 5)</td>
        <td><code>max-ion-charge</code>, double check the 5</td>
        <td>No</td>
      </tr>
      <tr>
        <td>max_precursor_charge</td><td>Parameter File</td>
        <td>set maximum precursor charge state to analyze (allowed max 9)</td>
        <td><code>max-precursor-charge</code>, why this and not use <code>precursor_charge</code>?</td>
        <td>No</td>
      </tr>
      <tr>
        <td>nucleotide_reading_frame</td><td>Parameter File</td>
        <td>0=proteinDB, 1-6, 7=forward three, 8=reverse three, 9=all six</td>
        <td><code>nucleotide-reading-frame</code>, what does 1-6, 7 mean?</td>
        <td>No</td>
      </tr>
      <tr>
        <td>clip_nterm_methionine</td>
        <td>Parameter File</td>
        <td>0=leave sequences as-is; 1=also consider sequence w/o N-term methionine</td>
        <td>No</td>
      </tr>
      <tr>
        <td>minimum_peaks</td><td>Parameter File</td>
        <td>minimum num. of peaks in spectrum to search (default 5)</td>
        <td><code>min-peaks</code></td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>minimum_intensity</td>
        <td>Parameter File</td>
        <td>minimum intensity value to read in</td>
        <td><code>minimum-intensity</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>remove_precursor_peak</td>
        <td>Parameter File</td>
        <td>0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD)</td>
        <td><code>remove-precursor-peak</code></td>
        <td>No</td>
      </tr>
      <tr>
        <td>remove_precursor_tolerance</td>
        <td>Parameter File</td>
        <td> +- Da tolerance for precursor removal</td>
        <td><code>remove-precursor-tolerance</code></td>
        <td>No</td>
      </tr>
      <tr>
          <td colspan="7"><b>additional modifications</b></td>
      </tr>
      <tr>
          <td>variable_C_terminus</td>
          <td>Parameter File</td>
          <td></td>
          <td></td>
          <td>No</td>
      </tr>
      <tr>
          <td>variable_N_terminus</td>
          <td>Parameter File</td>
      </tr>
      <tr>
          <td>variable_C_terminus_distance</td>
          <td>Parameter File</td>
          <td>-1=all peptides, 0=protein terminus, 1-N = maximum offset from C-terminus</td>
          <td></td>
      </tr>
      <tr>
          <td>variable_N_terminus_distance</td>
          <td>Parameter File</td>
          <td>-1=all peptides, 0=protein terminus, 1-N = maximum offset from N-terminus</td>
          <td></td>  
      </tr>
      <tr>
           <td>add_Cterm_peptide</td>
           <td>Parameter File</td>
      </tr>
      <tr>
           <td>add_Nterm_peptide</td>
           <td>Parameter File</td>
      </tr>
      <tr>
           <td>add_Cterm_protein</td>
           <td>Parameter File</td>
      </tr>
      <tr>
           <td>add_Nterm_protein</td>
           <td>Parameter File</td>
      </tr>
      <tr>
            <td>add_G_glycine</td>
            <td>Parameter File</td>
            <td>added to G - avg.  57.0513, mono.  57.02146</td>
            <td><code>G</code></td>
      </tr>
      <tr>
            <td>add_A_alanine = 0.0000</td>
            <td>Parameter File</td>
            <td>added to A - avg.  71.0779, mono.  71.03711</td>
            <td><code>A</code></td>
      </tr>
      <tr>
            <td>add_S_serine</td>
            <td>Parameter File</td>
            <td>added to S - avg.  87.0773, mono.  87.02303</td>
            <td><code>S</code></td>
      </tr>
      <tr>
            <td>add_P_proline</td>
            <td>Parameter File</td>
            <td>added to P - avg.  97.1152, mono.  97.05276</td>
            <td><code>P</code></td>
      </tr>
      <tr>
            <td>add_V_valine</td>
            <td>Parameter File</td>
            <td>added to V - avg.  99.1311, mono.  99.06841</td>
            <td><code>V</code></td>
      </tr>
      <tr>
            <td>add_T_threonine</td>
            <td>Parameter File</td>
            <td>added to T - avg. 101.1038, mono. 101.04768</td>
            <td><code>T</code></td>
      </tr>
          
      <tr>
            <td>add_C_cysteine</td>
            <td>Parameter File</td>
            <td>added to C - avg. 103.1429, mono. 103.00918</td>
            <td><code>C</code></td>
      </tr>
      <tr>
            <td>add_L_leucine</td>
            <td>Parameter File</td>
            <td>added to L - avg. 113.1576, mono. 113.08406</td>
            <td><code>L</code</td>
      </tr>
      <tr>
            <td>add_I_isoleucine</td>
            <td>Parameter File</td>
            <td>added to I - avg. 113.1576, mono. 113.08406</td>
            <td><code>I</code></td>
      </tr>
      <tr>
            <td>add_N_asparagine</td>
            <td>Parameter File</td>
            <td>added to N - avg. 114.1026, mono. 114.04293</td>
            <td><code>N</code></td>
      </tr>
      <tr>
            <td>add_D_aspartic_acid</td>
            <td>Parameter File</td>
            <td>added to D - avg. 115.0874, mono. 115.02694</td>
            <td><code>D</code></td>
      </tr>
      <tr>
            <td>add_Q_glutamine</td>
            <td>Parameter File</td>
            <td>added to Q - avg. 128.1292, mono. 128.05858</td>
            <td><code>Q</code></td>
      </tr>
      <tr>
            <td>add_K_lysine</td>
            <td>Parameter File</td>
            <td>added to K - avg. 128.1723, mono. 128.09496</td>
            <td><code>K</code></td>
      </tr>
      <tr>
            <td>add_E_glutamic_acid</td>
            <td>Parameter File</td>
            <td>added to E - avg. 129.1140, mono. 129.04259</td>
            <td><code>E</code></td>
      </tr>
      <tr>
            <td>add_M_methionine</td>
            <td>Parameter File</td>
            <td>added to M - avg. 131.1961, mono. 131.04048</td>
            <td><code>M</code></td>
      </tr>
      <tr>
            <td>add_O_ornithine</td>
            <td>Parameter File</td>
            <td>added to O - avg. 132.1610, mono  132.08988</td>
            <td><code>O</code></td>
      </tr>
      <tr>
            <td>add_H_histidine</td>
            <td>Parameter File</td>
            <td>added to H - avg. 137.1393, mono. 137.05891</td>
            <td><code>H</code></td>
      </tr>
      <tr>
            <td>add_F_phenylalanine</td>
            <td>Parameter File</td>
            <td>added to F - avg. 147.1739, mono. 147.06841</td>
            <td><code>F</code></td>
      </tr>
      <tr>
            <td>add_R_arginine</td>
            <td>Parameter File</td>
            <td>added to R - avg. 156.1857, mono. 156.10111</td>
            <td><code>R</code></td>
      </tr>
      <tr>
            <td>add_Y_tyrosine</td>
            <td>Parameter File</td>
            <td>added to Y - avg. 163.0633, mono. 163.06333</td>
            <td><code>Y</code></td>
      </tr>
      <tr>
            <td>add_W_tryptophan</td>
            <td>Parameter File</td>
            <td>added to W - avg. 186.0793, mono. 186.07931</td>
            <td><code>W</code></td>
      </tr>
      <tr>
            <td>add_B_user_amino_acid</td>
            <td>Parameter File</td>
            <td>added to B - avg.   0.0000, mono.   0.00000</td>
            <td><code>B</code></td>
      </tr>
      <tr>
            <td>add_J_user_amino_acid</td>
            <td>Parameter File</td>
            <td>added to J - avg.   0.0000, mono.   0.00000</td>
            <td><code>J</code></td>
      </tr>
      <tr>
            <td>add_U_user_amino_acid</td>
            <td>Parameter File</td>
            <td>added to U - avg.   0.0000, mono.   0.00000</td>
            <td><code>U</code></td>
      </tr>
      <tr>
            <td>add_X_user_amino_acid</td>
            <td>Parameter File</td>
            <td>added to X - avg.   0.0000, mono.   0.00000</td>
            <td><code>X</code></td>
      </tr>
      <tr>
            <td>add_Z_user_amino_acid</td>
            <td>Parameter File</td>
            <td>added to Z - avg.   0.0000, mono.   0.00000</td>
            <td><code>Z</code></td>
      </tr>
</table>
-->


</note>

<hr></hr>


<p>
<a href="crux.html">Crux home</a></p>

</blockquote>
</body>
</html>
