<html>
<title>Crux tandem mass spectrometry analysis software</title>
<link href="./crux.css" rel="styleSheet" type="text/css">
</head>
<body>

<H1 ALIGN=CENTER>Crux tandem mass spectrometry analysis software</H1>

<p>
Crux is a software toolkit for tandem mass spectrometry analysis, with
a focus on peptide identification.</p>

<p>
Crux is provided as a single executable that supports the following
commands:</p>

<table>

<tr>
<td colspan="2"><b>Primary commands</b></td>
</tr>

<tr>
<td>
<a href="crux-create-index.html">create-index</a></td>
<td>
Create an index for all peptides in a fasta file.</td>
</tr>

<tr>
<td>
<a href="crux-search-for-matches.html">search-for-matches</a></td>
<td>
Search a collection of spectra against a sequence database, returning
a collection of peptide-spectrum matches (PSMs).</td>
</tr>

<tr>
<td>
<a href="compute-q-values.html">compute-q-values</a></td>
<td>
Assign a q-value, which is a statistical confidence measure that
accounts for multiple testing, to each PSM in a given set.</td></tr>

<tr>
<td>
<a href="q-ranker.html">q-ranker</a></td>
<td>
Analyze a collection of PSMs using the Q-ranker algorithm.</td></tr>

<tr>
<td colspan="2"><b>Utilities</b></td>
</tr>

<tr>
<td>
<a href="crux-predict-peptide-ions.html">predict-peptide-ions</a></td>
<td>
Given a peptide and a charge state, predict the ions according to the
provided options.</td>
</tr>

<tr>
<td>
<a href="crux-generate-peptides.html">generate-peptides</a></td>
<td>
Given protein sequencs in a fasta file return a list of peptides
fitting the specified criteria.</td>
</tr>

<tr>
<td>
<a href="crux-get-ms2-spectrum.html">get-ms2-spectrum</a></td>
<td>
Extract an MS-MS spectrum with a specified scan number from a
<code>.ms2</code> file.</td>
</tr>

<tr>
<td>
<a href="spectacle.html">spectacle</a></td>
<td>
Create an image file of an annotated spectrum.  Unlike the other
commands, spectacle is a stand-alone program.</td>
</tr>

</table>

<h2>File formats</h2>

<ul>

<li>
MS2 spectrum files</li>

<li>
FASTA sequence files</li>

<li>
Peptide database index files (binary)</li>

<li>
Binary CSM files for representing PSMs.</li>

<li>
Text SQT files for representing PSMs.</li>

<li>
Text feature files for representing PSMs.</li>

</ul>

<p>
Download crux <a href="download/index.html">here</a> once you agree to
the license agreement and obtain a
password <a href="http://depts.washington.edu/ventures/UW_Technology/Express_Licenses/crux.php">here</a>.
Crux source code is freely available for academic use and can be
licensed for commercial use.</p>

<p>
If you use Crux in your research, please cite</p>

<blockquote>
Christopher Y. Park, Aaron A. Klammer, Lukas K&auml;ll, Michael
J. MacCoss and William Stafford
Noble.  <a href="http://noble.gs.washington.edu/papers/park2008rapid.html">"Rapid
and accurate peptide identification from tandem mass spectra."</a>
<i>Journal of Proteome Research.</i> 7(7):3022--3027, 2008.</li>
</blockquote>

<p>
A tutorial describing how to install and use Crux can be
found <a href="home-tutorial.html">here</a>.</p>

<p>
Crux can interface with the Percolator algorithm, which can be
downloaded
separately <a href="http://noble.gs.washington.edu/proj/percolator/">here</a>.
<a href="">This page</a> describes how to use Crux with
Percolator.</p>

<p>
<a href="release-notes.html">Release notes.</a></p>

<hr></hr>

<center>
<p>
Crux is written in C by Chris Park, Aaron Klammer, Barbara Frewen and
Charles Grant under the supervision
of <a href="http://noble.gs.washington.edu/~noble">Prof. William
Stafford Noble</a> in the Department of Genome Sciences at the
University of Washington, Seattle.</p>

<p>
Maintenance and development of Crux is funded by the National
Institute of Biomedical Imaging and Bioengineering grant NIH/NIBIB R01
EB007057.</p></center>

<hr></hr>

<center>
<img src="nibib.gif">
<img src="UWlogo150p.gif">
</center>

<p>
Please send comments and questions to FIXME.</p>

</body>
</html>
