<html>
<head>
<title>crux comet</title>
<link href="./crux.css" rel="styleSheet" type="text/css">
</head>
<body>
<blockquote>

<H1 ALIGN=CENTER>crux comet</H1>
<!-- open source warning
<h2>
You are using the open source version of Crux.
Due to intellectual property issues,
we are unable to provide database search functionality in this version.
To obtain a license for the fully functional version of Crux that includes
the database search tools please visit 
<a href="http://depts.washington.edu/ventures/UW_Technology/Express_Licenses/crux.php">UW TECHTRANSFER</a>.
</h2>
open source warning -->
<B>Usage:</B>

<blockquote>
<code>crux comet [options]
&lt;ms2&nbsp;input&nbsp;filename&gt; &lt;protein&nbsp;input&gt;
</code>
</blockquote>

<B>Description:</B>

<blockquote>
<p>
This command searches a protein database with a set of spectra, using
an algorithm similar to that employed by the SEQUEST database search
software.  This command differs from the <code>crux
search-for-matches</code> command in the following respects:</p>
<ul>
<li>
Candidate peptides (i.e., peptides that lie within the user-specified
precursor window) are first ranked with the SEQUEST Sp score.  The top
500 matches are then re-scored using XCorr.</li>
<li>
The program will not compute p-values using the Weibull empirical
curve fitting procedure.</li>
<li>
By default, the theoretical spectrum used to compute XCorr includes
two flanking peaks on either side of each b- and y-ion.  These can be
turned off using the <code>use-flanking-peaks</code> parameter.</li>
<li>
The program produces <a href="sqt-format.html">SQT</a> format output
files.</li>
</ul>

<p>
Similar to <code>crux search-for-matches</code>, the input protein
database may either be in FASTA format or it may be a binary index
created by <code>crux create-index</code>.  Using an index will
typically yield much faster search speeds.</p>

<p>
This command handles modifications in the same way
as <a href="crux-search-for-matches.html"><code>crux
search-for-matches</code></a>.</p>

</blockquote>

<p>
<B>Input:</B>

<UL>

<LI>
&lt;ms2 &gt; &ndash; The name of the file (in MS2 format) from which
  to parse the spectra. </li>

<LI>
&lt;protein input&gt; &ndash; 
  The name of the file in fasta format or the directory containing a
  protein index from which to retrieve proteins and peptides.</li>

</UL>

<B>Output:</B>

<p>
The program writes files to the folder <code>crux-output</code> by
default.  The name of the output folder can be set by the user using
the <code>--output-dir</code> option.  The following files will be
created:</p>
<ul>

<li>
<b>sequest.params.txt</b>: a file containing the name and value of all
parameters/options for the current operation.  Not all parameters in
the file may have been used in the operation.  The resulting file can
be used with the <code>--parameter-file</code> option for other crux
programs.
</li>

<li>
<b>sequest.target.sqt</b>: an <a href="sqt-format.html">SQT</a> file
containing the PSMs.</li>

<li>
<b>sequest.target.pep.xml</b>

<li>
<b>sequest.log.txt</b>: a log file containing a copy of all messages
that were printed to stderr.

</ul>

<p>
If decoys are enabled using <code>--num-decoys-per-target</code>, then
search.decoy.sqt and search.decoy.pep.xml are also produced.</p>

<B>Options:</B>
<UL>

<li>
<code>--fileroot &lt;string&gt;</code> &ndash; 
The <code>fileroot</code> string will be added as a prefix to all
output file names. Default = none.</li>

<li>
<code>--output-dir &lt;filename&gt;</code> &ndash; 
The name of the directory where output files will be created.
Default = crux-out.</li>


<li>
<code>--overwrite &lt;T|F&gt;</code> &ndash;
Replace existing files if true (T) or fail when trying to overwrite a
file if false (F). Default =  F.</li>

<li>
<code>--decoys &lt;string&gt;</code> &ndash; Include a decoy version
of every peptide by shuffling or reversing the target sequence.
Possible values are <code>none, reverse, protein-shuffle,
peptide-shuffle</code>. Use 'reverse' to reverse each protein sequence,
'protein-shuffle' to shuffle each protein sequence, or 'peptide-shuffle'
to shuffle the sequence between enzyme cleavage sites, leaving the
termini in place. Use 'none' for no decoys.  Default=protein-shuffle. 

<LI>
<code>--num-decoys-per-target &lt;n&gt;</code> &ndash; 
Specify the number of decoy peptides to search for every target
peptide searched. Control where the decoys are returned (to what
files) with <code>--decoy-location</code>. At least one decoy set (in
its own file) is required to run the algorithm 'percolator' in a
subsequent crux run.  Default = 2. </li>

<li>
<code>--decoy-location &lt;target-file | one-decoy-file |
separate-decoy-files&gt;</code> &ndash; 
File(s) in which decoy results are returned.  Only applies
when <code>num-decoys-per-target</code> is not zero.  Use
'target-file' to mix target and decoy psms in one file. Use
'one-decoy-file' to print target psms to one file and all decoys to a
separate file.  Use 'separate-decoy-files' to print one .txt file for
each decoy set.  (<code>crux percolator</code> accepts up to two
search.decoy.txt files. <code>crux q-ranker</code> accepts only one
search.decoy.txt file.) Default = separate-decoy-files.


<LI>
<code>--spectrum-min-mass &lt;float&gt;</code> &ndash; 
The lowest spectrum m/z to search in the ms2 file. Default = 0.0

<LI>
<code>--spectrum-max-mass &lt;float&gt;</code> &ndash; 
The highest spectrum m/z to search in the ms2 file. Default = no maximum.

<LI> 
<code>--spectrum-charge &lt;1|2|3|all&gt;</code> &ndash; 
The spectrum charges to search. With 'all' every spectrum will be
searched and spectra with multiple charge states will be searched once
at each charge state.  With 1, 2, or 3 only spectra with that charge
will be searched. Default = all. 

<LI>
<code>--scan-number &lt;number range&gt;</code> &ndash;
A single scan number or a range of numbers to be searched. Range
should be specified as 'first-last' which will include scans 'first'
and 'last'.  Default = search all spectra.

<LI><code>--parameter-file &lt;filename&gt;</code> &ndash; 
A file containing command-line or additional parameters.  See
the <a href="parameter-file.html">parameter documentation</a> page
for details.  Default = no parameter file.

<LI>
<code>--verbosity &lt;0-100&gt;</code> &ndash; 
Specify the verbosity of the current processes. Each level prints the
following messages, including all those at lower verbosity levels:
0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on
the progress of execution, 40-more progress information, 50-debug
info, 60-detailed debug info. Default = 30.

</UL>

<B>Parameter file options:</B>

<ul>

<LI>
<code>--min-peaks &lt;int&gt;</code> &ndash;
The minimum number of peaks a spectrum must have in order to be
searched.  Default = 20.

<li>
<code>fragment-mass &lt;average|mono&gt;</code> &ndash; 
Which isotopes to use in calculating fragment ion mass (average,
mono). Default = mono.</li>

<li>
<code>use-flanking-peaks &lt;T|F&gt;</code> &ndash;
Turn on or off the peaks flanking the b/y ions.  For <code>crux 
search-for-matches</code>, default = F; for <code>crux
  comet</code>, default = T; for <code> crux
search-for-xlinks</code>, default = T.</li>
 
<li>
<code>precursor-window &lt;float&gt;</code> &ndash; Tolerance used for
matching peptides to spectra.  Peptides must be within +/-
'<code>precursor-window</code>' of the spectrum value.  Definition of
precursor window units depends
upon <code>precursor-window-type</code>. Default = 3.0.</li>

<li>
<code>precursor-window-type &lt;mass|mz|ppm&gt;</code> &ndash; Specify
the units for the window that is used to select peptides around the
precursor mass location (mass, mz, ppm).  The magnitude of the window
is defined by the <code>precursor-window</code> option, and candidate
peptides must fall within this window.  For the <code>mass</code>
window-type, the spectrum precursor singly charged mass (m+h) is
converted to mass, and the window is defined as that mass
&plusmn; <code>precursor-window</code>. If the m+h value is not
available, then the mass is calculated from the precursor
mass-to-charge (m/z) and provided charge.  The peptide mass is
computed as the sum of the average amino acid masses plus 18 Da for
the terminal OH group.  The <code>mz</code> window-type calculates the
window as spectrum precursor m/z
&plusmn; <code>precursor-window</code> and then converts the resulting
m/z range to the peptide mass range using the precursor charge.  For
the parts-per-million (<code>ppm</code>) window-type, the spectrum
mass is calculated as in the <code>mass</code> type.  The lower bound
of the mass window is then defined as the spectrum mass / (1.0 +
(<code>precursor-window</code> / 1000000)) and the upper bound is
defined as spectrum mass / (1.0 - (<code>precursor-window</code> /
1000000)).  Default = mass.</li>

<li>
<code>use-mstoolkit &lt;T|F&gt;</code> &ndash;
Use MSToolkit to parse the spectra. Required for parsing spectra from
.mzXML files. Default = F.

<li>
<code>top-match &lt;int&gt;</code> &ndash; 
The number of psms per spectrum written to the output files. Default =
5.

<li>
<code>max-rank-preliminary &lt;int&gt;</code> &ndash;
The number of PSMs to score with XCorr after ranking PSMs by Sp.
Default = 500.

<li>

<code>mod &lt;mass&nbsp;change&gt;:&lt;aa&nbsp;list&gt;:&lt;max&nbsp;per&nbsp;peptide&gt;:&lt;prevents&nbsp;cleavage&gt;:&lt;prevents&nbsp;cross-link&gt;
</code> &ndash; 
Consider modifications on any amino acid in aa list with at most
max-per-peptide in one peptide.  This parameter may be included
with different values multiple times so long as the total number of
<code>mod</code>, <code>cmod</code>, and <code>nmod</code> parameters
does not exceed 11.  The <code>prevents&nbsp;cleavage</code> and 
<code>prevents&nbsp;cross-link</code> are T/F optional arguments for 
describing whether the modification prevent enzymatic cleavage or 
cross-linking respectively. The same modifications must be given for any
post-search process (<code>crux compute-q-values</code>, <code>crux
q-ranker</code>, <code>crux percolator</code>). Default = no
variable modifications.

<li>
<code>cmod &lt;mass change&gt;:&lt;max distance from protein
C-terminus&gt;</code> &ndash;
Consider modifications on the C-terminus of any peptide whose
C-terminus is no more than max-distance residues from the protein
C-terminus.  Use -1 to consider the C-terminus of all peptides
regardless of position in the protein.  This parameter may be included
with different values multiple times so long as the total number
of <code>mod</code>, <code>cmod</code>, and <code>nmod</code>
parameters does not exceed 11. The same modifications must be given
for any post-search process (<code>crux
  compute-q-values</code>, <code>crux q-ranker</code>, <code>crux
  percolator</code>). Default = no c-terminal modifications.

<li>
<code>nmod &lt;mass change&gt;:&lt;max distance from protein N-terminus&gt;
&ndash;</code> 
Consider modifications on the N-terminus of any peptide whose
N-terminus is no more than max-distance residues from the protein
N-terminus.  Use -1 to consider the N-terminus of all peptides regardless
of position in the protein. This parameter may be included with
different values multiple times so long as the total number of
<code>mod</code>, <code>cmod</code>, and <code>nmod</code> parameters
does not exceed 11. The same modifications must be given for any
post-search process (<code>crux compute-q-values</code>, <code>crux 
q-ranker</code>, <code>crux percolator</code>). Default = no
n-terminal modifications.

<li>
<code>cmod-fixed &lt;mass change&gt&ndash;</code>
Add a modification of the given mass change to the C-terminus of every
peptide.

<li>
<code>nmod-fixed &lt;mass change&gt&ndash;</code>
Add a modification of the given mass change to the N-terminus of every
peptide.

<li>
<code>max-mods &lt;int&gt;</code> &ndash;
The maximum number of modifications that can be applied to a single
peptide. Default = no limit.

<li>
<code>max-aas-modified &lt;int&gt;</code> &ndash;
The maximum number of modified amino acids that can appear in one
peptide.  Each aa can be modified multiple times. Default = no limit.

<li>
<code>mod-mass-format &lt;mod-only|total|separate&gt;<code> &ndash;
Specify how sequence modifications are reported in various ouptut
files.  Each modification is reported as a number enclosed in square
braces following the modified reside; however, the number may
correspond to one of three different masses: (1) 'mod-only' reports
the value of the mass shift induced by the modification; (2) 'total'
reports the mass of the residue with the modification (residue mass
plus modification mass); (3) 'separate' is the same as 'mod-only', but
multiple modifications to a single amino acid are reported as a
comma-separated list of values. For example, suppose amino
acid <i>D</i> has an unmodified mass of 115 as well as two
modifications of masses +14 and +2.  In this case, the amino acid
would be reported as D[16] with 'mod-only', D[131] with 'total', and
D[14,2] with 'separate'.</li>

<li>
<code>precision &lt;int&gt;</code> &ndash;
Set the precision (number of significant digits) for scores
written to text files. Default = 8.

<li>
<code>print-search-progress &lt;int&gt;</code> &ndash;
Show search progress by printing every n spectra searched. Set to
0 to show no search progress.  Default = 10.

<li>
<code>reverse-sequence &lt;T|F&gt;</code> &ndash;
Generate decoy sequences by reversing the peptide rather than by
shuffling.  The first and last residues of the sequence are not
changed.  If the target sequence is a palindrome (the same when
reversed), then the decoy will be generated by shuffling and a note to
that effect will be printed at verbosity level 40 (DETAILED
INFO). Default = generate decoys by shuffling.

<p>
<i>NOTE: the following parameters are also used when creating an index
and must be compatible with any index used.</i></p>

<LI>
<code>min-mass &lt;float&gt;</code> &ndash; 
The minimum neutral mass of the peptides to place in the index. Default
= 200.

<LI>
<code>max-mass &lt;float&gt;</code> &ndash; 
The maximum neutral mass of the peptides to place in index. Default = 7200.
	
<LI>
<code>min-length &lt;int&gt;</code> &ndash; 
The minimum length of the peptides to place in the index. Default = 6.
	
<LI>
<code>max-length &lt;int&gt;</code> &ndash; 
The maximum length of the peptides to place in the index. Default = 50.

<LI>
<code>enzyme &lt;trypsin|chymotrypsin|elastase|clostripain|cyanogen-bromide|idosobenzoate|proline-endopeptidase|staph-protease|modified-chymotrypsin|elastase-trypsin-chymotrypsin|no-enzyme&gt;
</code> &ndash;
Enzyme to use for <i>in silico</i> digestion of protein
sequences. Used in conjunction with the <code>digestion</code>
and <code>missed-cleavages</code> options. Use 'no-enzyme' for
non-specific digestion.  Digestion rules are as follows: enzyme name
[cuts after one of these residues]|{but not before one of these
residues}.  trypsin [RK]|{P}, elastase [ALIV]|{P}, chymotrypsin
[FWY]|{P}, clostripain [R]|[], cyanogen-bromide [M]|[], iodosobenzoate
[W]|[], proline-endopeptidase [P]|[], staph-protease [E]|[],
modified-chymotrypsin [FWYL]|{P}, elastase-trypsin-chymotrypsin
[ALIVKRWFY]|{P},aspn [X]|[D] (cuts after any residue and before
D). Default = trypsin.

<li>
<code>custom-enzyme &lt;residues before cleavage | residues after cleavage
&gt;</code> &ndash
Specify rules for <i>in silico</i> digestion of protein sequences.
Overrides the <code>enzyme</code> option.  Two lists of residues are
given enclosed in square brackets or curly braces and separated by a
|. The first list contains residues required/prohibited before the
cleavage site and the second list is residues after the cleavage site.
If the residues are required for digestion, they are in square
brackets, '[' and ']'.  If the residues prevent digestion, then they
are enclosed in curly braces, '{' and '}'.  Use X to indicate all
residues.  For example, trypsin cuts after R or K but not before P
which is represented as <code>[RK]|{P}</code>.  AspN cuts after any
residue but only before D which is represented as <code>[X]|[D]</code>.

<LI>
<code>digestion &lt;full-digest|partial-digest&gt;</code> &ndash;
Degree of digestion used to generate peptides . Either both ends
(full-digest) or at least one end (partial-digest) of a peptide must
conform to enzyme specificity rules. Used in conjunction with
the <code>enzyme</code> or <code>custom-enzyme</code> option
when <code>enzyme</code> is not set to to 'no-enzyme'.   Default
full-digest.

<LI>
<code>missed-cleavages &lt;T|F&gt;</code> &ndash;
Allow missed cleavage sites within a peptide.  When used with 
<code>enzyme</code> and set to true, includes peptides containing
one or more potential cleavage sites. Default = F.

<LI>
<code>isotopic-mass &lt;average|mono&gt;</code> &ndash; 
Specify the type of isotopic masses to use when calculating the
peptide mass. Default = average.
   
<li>
<code>&lt;A-Z&gt &lt;float&gt;</code> &ndash; 
Specify static modifications.  This is a mass change applied to the
given amino acid (in single-letter-code A thru Z) for every peptide in
which it occurs. Use the <code>mod</code> option for generating
peptides both with and without the mass change.  Default C=57.
</ul>

<B>Options/arguments not in Comet</B>
<ul>
<li></li>
</ul>

<B>New Parameter file options</B>
<ul>
<li>
<code>num-threads &lt;int&gt;</code> &ndash;
</li>
<li>
<code>isotope-windows &lt;list of integers&gt;</code> &ndash;
</li>

</ul>

<HR></hr>

      <P>
        <A HREF="crux.html">Crux home</A>
      </P>

      
      <note>

 

<p>
To do:</p>

<ul>
<li>
<li>
Figure out how to handle MSToolkit issues (Comet has its own copy, we need
it for harklor/bullseye.)
</li>

</ul>
<p>Argument parameter mappings</p>
<br>
    Comet usage:  ../comet.exe [options] <input_files>[:range]<br>
    The optional [:range] parameter specifies a scan range to search<br>
    <table border="1">
      <tr>
        <td>Comet parameter</td>

        <td>Parameter File/Command Line</td>
        <td>Parameter Description</td>
        <td>Crux Parameter</td>
        <td>Exists in Crux</td>
        <td>Exists in Comet</td>
      </tr>
      <tr>
        <td>-p</td>
        <td>Command Line</td>

        <td>print out comet.params.new</td>
        <td></td>
      </tr>
      <tr>
        <td>-P params</td>
        <td>Command Line</td>
        <td>specify an alternate parameter file</td>
        <td>Override this with our generated parameter file</td>
        <td>N/A</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>-N name</td>
        <td>Command Line</td>
        <td>specifiy an alterate output base name</td>
        <td>Override this with crux-output</td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>-D dbase</td>
        <td>Command Line</td>
        <td>specify a sequence database, overrideing entry in parameters file</td>
        <td><code>protein database</code></td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>

      <tr>
        <td>[:range]</td>
        <td>Command Line</td>
        <td>Specify range of scans to search</td>
        <td><code>--scan-number</code</td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>

        <td><code>database_name</code></td>
        <td>Parameter File</td>
        <td>Fasta file of database</td>
        <td><code>protein database</code></td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>decoy_search</td>
        <td>Parameter File</td>
        <td>0= no, 1=concatenated, 2=separate search</td>
        <td><code>--decoy-location</code>,<code>--num-decoys-per-target</code></td>
        <td>Yes*</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>num_threads</td>
        <td>Parameter File</td>
        <td>0=poll CPU to set threads; </td>
        <td><code>--num-threads</code></td>
        <td>No</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>peptide_mass_tolerance</td>
        <td>Parameter File</td>
        <td>Mass tolerance to use when selecting candidate peptides</td>
        <td><code>--precursor-window</td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>peptide_mass_units</td>
        <td>Parameter File</td>
        <td>Units to use when selecting peptides, 0=amu, 1=mmu, 2=ppm</td>
        <td><code>--precursor-window-type</code></td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>mass_type_parent</td>
        <td>Parameter File</td>
        <td>average or monoisotopic mass for calculating peptide mass 0=average mass, 1=monoisotopic mass</td>
        <td><code>--isotopic-mass</code>
        <td>Yes</td>
        <td>Yes</td>
      </tr>

      <tr>
        <td>mass_type_fragment</td>
        <td>Parameter File</td>
        <td>average or monoisotopic mass for calculating fragment masses</td>
        <td><code>--fragment-mass</code>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>

        <td>precursor_tolerance_type</td>
        <td>Parameter File</td>
        <td>0=MH+, 1=precursor m/z</td>
        <td>???</td>
      </tr>
      <tr>
        <td>isotope_error</td>
        <td>Parameter File</td>
        <td>Search in small windows that are +/- 1 Neutron Mass away from precursor mass?</td>
        <td><code>--isotope-windows</td>
        <td>No</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>search_enzyme_number</td>
        <td>Parameter File</td>
        <td>Enzyme to use when digesting peptides</td>
        <td><code>--enzyme</code></td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>num_enzyme_termini</td>
        <td>Parameter File</td>
        <td>selecting semi-digestion, full digestion, and others valid values are 1 (semi-digested), 2 (fully digested, default), 8 N-term, 9 C-term</td>
        <td><code></code></td>
      </tr>
      <tr>
        <td>allowed_missed_cleavage</td>
        <td>Parameter File</td>
        <td>Number of missed cleavages allowed (max 5)</td>
        <td><code>--missed-cleaveages</code></td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>variable_modX (X=1-6)</td>
        <td>Parameter File</td>
        <td>Variable modification support</td>
        <td>Supported using mod</td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>max_variable_mods_in_peptide</td>
        <td>Parameter File</td>
        <td>Maximum number of mods to consider in peptide</td>
        <td><code>--max-mods</code></td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>fragment_bin_tol</td>
        <td>Parameter File</td>
        <td>bin width to use on fragment ions</td>
        <td>mz-bin-width</td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>fragment_bin_offset</td>
        <td>Parameter File</td>
        <td>offset to start the binning</td>
        <td>mz-bin-offset</td>
        <td>Yes</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>theoretical_fragment_ions</td>
        <td>Parameter File</td>
        <td>??? 0=default peak shape, 1=M peak only</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>use_A_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use a-ions</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>use_B_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use a-ions</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>use_C_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use a-ions</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>        
      <tr>
        <td>use_X_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use a-ions</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>
      <tr>
        <td>use_Y_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use a-ions</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>

      <tr>
        <td>use_Z_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use a-ions</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>
       <tr>
        <td>use_NL_ions</td>
        <td>Parameter File</td>
        <td>Whether or not to use NH3/H2O neutral loss peaks</td>
        <td></td>
        <td>No</td>
        <td>Yes</td>
      </tr>     
      <tr>
        <td>output_sqtstream</td><td>Parameter File</td>
        <td>0=no, 1=yes  write sqt to standard output</td>
      </tr>
      <tr>
        <td>output_sqtfile</td><td>Parameter File</td>
        <td>0=no, 1=yes  write sqt file</td>
      </tr>
      <tr>
        <td>output_pepxmlfile</td><td>Parameter File</td>
        <td>0=no, 1=yes  write pep.xml file</td>
      </tr>
      <tr>
        <td>output_outfiles</td><td>Parameter File</td>
        <td>0=no, 1=yes  write .out files</td>
      </tr>
      <tr>
        <td>print_expect_score</td><td>Parameter File</td>
        <td>to replace Sp with expect in out & sqt</td>
      </tr>
      <tr>
        <td>num_output_lines</td><td>Parameter File</td>
        <td>num peptide results to show</td>
      </tr>
      <tr>
        <td>show_fragment_ions</td><td>Parameter File</td>
        <td> 0=no, 1=yes for out files only</td>
      </tr>
      <tr>
        <td>sample_enzyme_number</td><td>Parameter File</td>
        <td>Sample enzyme which is possibly different than the one applied to the search.<br>
          Used to calculate NTT & NMC in pepXML output (default=1 for trypsin).</td>
      </tr>
      <tr>
         <td colspan="6"><B>mzXML parameters</B></td>
      </tr>
      <tr>
        <td>scan_range</td><td>Parameter File</td>
        <td>start and scan scan range to search; 0 as 1st entry ignores parameter</td>
      </tr>
      <tr>
        <td>precursor_charge</td><td>Parameter File</td>
        <td>precursor charge range to analyze; does not override mzXML charge; 0 as 1st entry ignores parameter</td>
      </tr>
      <tr>
         <td>ms_level</td><td>Parameter File</td>
         <td>MS level to analyze, valid are levels 2 (default) or 3</td>
      </tr>
      <tr>
        <td>activation_method</td><td>Parameter File</td>
        <td>activation method; used if activation method set; allowed ALL, CID, ECD, ETD, PQD, HCD, IRMPD</td>
      </tr>
      <tr>
        <td colspan="6"><B>misc parameters</td>
      </tr>
      <tr>
        <td>digest_mass_range</td><td>Parameter File</td>
        <td>MH+ peptide mass range to analyze</td>
      </tr>
      <tr>
        <td>num_results</td><td>Parameter File</td>
        <td>number of search hits to store internally</td>
      </tr>
      <tr>
        <td>skip_researching</td><td>Parameter File</td>
        <td>for '.out' file output only, 0=search everything again (default), 1=don't search if .out exists</td>
      </tr>
      <tr>
        <td>max_fragment_charge</td><td>Parameter File</td>
        <td>set maximum fragment charge state to analyze (allowed max 5)</td>
      </tr>
      <tr>
        <td>max_precursor_charge</td><td>Parameter File</td>
        <td>set maximum precursor charge state to analyze (allowed max 9)</td>
      </tr>
      <tr>
        <td>nucleotide_reading_frame</td><td>Parameter File</td>
        <td>0=proteinDB, 1-6, 7=forward three, 8=reverse three, 9=all six</td>
      </tr>
      <tr>
        <td>clip_nterm_methionine</td><td>Parameter File</td>
        <td>0=leave sequences as-is; 1=also consider sequence w/o N-term methionine</td>
      </tr>
      <tr>
        <td>minimum_peaks</td><td>Parameter File</td>
        <td>minimum num. of peaks in spectrum to search (default 5)</td>
      </tr>
      <tr>
        <td>minimum_intensity</td><td>Parameter File</td>
        <td>minimum intensity value to read in</td>
      </tr>
      <tr>
        <td>remove_precursor_peak</td><td>Parameter File</td>
        <td>0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD)</td>
      </tr>
      <tr>
        <td>remove_precursor_tolerance</td><td>Parameter File</td>
        <td> +- Da tolerance for precursor removal</td>
      </tr>
      <tr><td>variable_C_terminus = 0.0</td><td>Parameter File</td></tr>
      <tr><td>variable_N_terminus = 0.0</td><td>Parameter File</td></tr>
      <tr><td>variable_C_terminus_distance</td><td>Parameter File</td>
          <td>-1=all peptides, 0=protein terminus, 1-N = maximum offset from C-terminus</td>
      </tr>
      <tr><td>variable_N_terminus_distance</td><td>Parameter File</td>
          <td>-1=all peptides, 0=protein terminus, 1-N = maximum offset from N-terminus</td>
      </tr>
      <tr><td>add_Cterm_peptide</td><td>Parameter File</td></tr>
      <tr><td>add_Nterm_peptide</td><td>Parameter File</td></tr>
      <tr><td>add_Cterm_protein</td><td>Parameter File</td></tr>
      <tr><td>add_Nterm_protein</td><td>Parameter File</td></tr>
      <tr><td>add_G_glycine</td><td>Parameter File</td>
          <td>added to G - avg.  57.0513, mono.  57.02146</td></tr>
      <tr><td>add_A_alanine = 0.0000</td><td>Parameter File</td>
          <td>added to A - avg.  71.0779, mono.  71.03711</td>
      </tr>
      <tr><td>add_S_serine</td><td>Parameter File</td>
          <td>added to S - avg.  87.0773, mono.  87.02303</td>
      </tr>
      <tr><td>add_P_proline</td><td>Parameter File</td>
          <td>added to P - avg.  97.1152, mono.  97.05276</td>
      </tr>
      <tr><td>add_V_valine</td><td>Parameter File</td>
          <td>added to V - avg.  99.1311, mono.  99.06841</td>
      </tr>
      <tr><td>add_T_threonine</td><td>Parameter File</td>
          <td>added to T - avg. 101.1038, mono. 101.04768</td></tr>
      <tr><td>add_C_cysteine</td><td>Parameter File</td>
          <td>added to C - avg. 103.1429, mono. 103.00918</td></tr>
      <tr><td>add_L_leucine</td><td>Parameter File</td>
          <td>added to L - avg. 113.1576, mono. 113.08406</td></tr>
      <tr><td>add_I_isoleucine</td><td>Parameter File</td>
          <td>added to I - avg. 113.1576, mono. 113.08406</td></tr>
      <tr><td>add_N_asparagine</td><td>Parameter File</td>
          <td>added to N - avg. 114.1026, mono. 114.04293</td></tr>
      <tr><td>add_D_aspartic_acid</td><td>Parameter File</td>
          <td>added to D - avg. 115.0874, mono. 115.02694</td></tr>
      <tr><td>add_Q_glutamine</td><td>Parameter File</td>
          <td>added to Q - avg. 128.1292, mono. 128.05858</td></tr>
      <tr><td>add_K_lysine</td><td>Parameter File</td>
          <td>added to K - avg. 128.1723, mono. 128.09496</td></tr>
      <tr><td>add_E_glutamic_acid</td><td>Parameter File</td>
          <td>added to E - avg. 129.1140, mono. 129.04259</td></tr>
      <tr><td>add_M_methionine</td><td>Parameter File</td>
          <td>added to M - avg. 131.1961, mono. 131.04048</td></tr>
      <tr><td>add_O_ornithine</td><td>Parameter File</td>
          <td>added to O - avg. 132.1610, mono  132.08988</td></tr>
      <tr><td>add_H_histidine</td><td>Parameter File</td>
          <td>added to H - avg. 137.1393, mono. 137.05891</td></tr>
      <tr><td>add_F_phenylalanine</td><td>Parameter File</td>
          <td>added to F - avg. 147.1739, mono. 147.06841</td></tr>
      <tr><td>add_R_arginine</td><td>Parameter File</td>
          <td>added to R - avg. 156.1857, mono. 156.10111</td></tr>
      <tr><td>add_Y_tyrosine</td><td>Parameter File</td>
          <td>added to Y - avg. 163.0633, mono. 163.06333</td></tr>
      <tr><td>add_W_tryptophan</td><td>Parameter File</td>
          <td>added to W - avg. 186.0793, mono. 186.07931</td></tr>
      <tr><td>add_B_user_amino_acid</td><td>Parameter File</td>
          <td>added to B - avg.   0.0000, mono.   0.00000</td></tr>
      <tr><td>add_J_user_amino_acid</td><td>Parameter File</td>
          <td>added to J - avg.   0.0000, mono.   0.00000</td></tr>
      <tr><td>add_U_user_amino_acid</td><td>Parameter File</td>
          <td>added to U - avg.   0.0000, mono.   0.00000</td></tr>
      <tr><td>add_X_user_amino_acid</td><td>Parameter File</td>
          <td>added to X - avg.   0.0000, mono.   0.00000</td></tr>
      <tr><td>add_Z_user_amino_acid</td><td>Parameter File</td>
          <td>added to Z - avg.   0.0000, mono.   0.00000</td></tr>
</table>

<p>
Changes we are proposing on the Comet end of things:</p>
<ul>
<li>
Have an option to compile Comet as a library in the with the main() replaced with comet_main().
</li>
</ul>

</note>


</blockquote>
</BODY>
</HTML>
