<html>
<title>crux percolator</title>
<link href="./crux.css" rel="styleSheet" type="text/css">
</head>
<body>
<blockquote>

<H1 ALIGN=CENTER>crux percolator</H1>

<p>
<b>Usage:</b></p>

<blockquote>
<code>crux percolator [-xmloutput pout.xml][other options] &lt;pin.xml&gt; </code>
</blockquote>

<p>
<b>Description:</b></p>

<p>
Percolator is a semi-supervised learning algorithm that dynamically
learns to separate target from decoy PSMs.  The algorithm is described
in this article:</p>

<blockquote>
Lukas K&auml;ll, Jesse Canterbury, Jason Weston, William Stafford
Noble and Michael
J. MacCoss.  <a href="http://noble.gs.washington.edu/papers/kall2007semi-supervised.html">"Semi-supervised
learning for peptide identification from shotgun proteomics
datasets."</a>
<i>Nature Methods</i>.  4(11):923-925, 2007.
</blockquote>

<p>
For more on q-values and posterior error probabilities (PEP), see the
documentaion for <a href=calibrate-scores.html>calibrate-scores</a>.
<p>
<span class=prog>Crux</span> now includes code from
<span class=prog>Percolator 1.05</span>.
The current version of Percolator can be downloaded
<a href="http://noble.gs.washington.edu/proj/percolator/">here</a>.
External versions of Percolator can be applied to the output of the 
<span class=prog>Crux</span> tools, but won't be used by the
<code>crux percolator</code> command.

<p>
Percolator requires as input two collections of PSMs, one set derived
from matching observed spectra against real ("target") peptides, and
a second derived from matching against "decoy" peptides.  Crux generates
these decoys on the fly.  Percolator will also accept a second set of
decoy PSMs and use one set in training and one in calculating
q-values.  Producing two sets of decoy PSMs is the search-for-matches
default behavior.</p>

<p>
The features used by percolator to represent each PSM are summarized
<a href="features.html">here</a>.</p>

<p>
<b>Input:</b></p>

<ul>
<LI>
<b>&lt;pin.xml&gt;</b> &ndash; PIN files are XML files for PIN format whose structure is defined by 
the schema percolator_in.xml. This file is generated by a convertor program called 
sqt2pin. Sqt2pin can convert (merge) target.sqt and decoy.sqt into a PIN file. </li>
<!--
<li>&lt;protein input&gt; &ndash; The name of the file in fasta format or the directory containing the protein index from which to retrieve proteins and peptides.
</li

<li>
&lt;search results directory&gt; &ndash; A folder in which all the PSM
result files are located.  The program looks for files produced by 
<code>crux search-for-matches</code> (ending in
'<code>.search.target.txt</code>' or '<code>.search.decoy.txt</code>')
<!--or by <code>crux sequest-search</code> (ending in
'<code>.sequest.target.txt</code>' or
'<code>.sequest.decoy.txt</code>').  All such files in the given
directory are analyzed jointly.  Note that the directory should not
contain results from both types of search algorithms.</li-->

</ul>

<B>Output:</B>
<p>  
Percolator writes files to the file <code>pout.xml</code> by default.
The path of the output file can be set by using the <code>--xmloutput</code> option. The user can use the  <code>
--results</code> parameter to output to a file in <code>crux-ouput</code> directory.
The following files will be created:</p>
<ul>
<li>
<b>percolator.params.txt</b>:
a file containing the name and value of all parameters/options for the current operation.  
Not all parameters in the file may have been used in the operation.  
The resulting file can be used with the <code>--parameter-file</code>
option for other crux programs.
</li>

<li>
<b>percolator.target.output.xml</b>: a xml file containing the PSMs.
See <note>here</note> for a list of the
fields.</li>

<li>
<b>percolator.target.pep.xml</b>: a file containing the PSMs in pepxml format.
See <a href="http://tools.proteomecenter.org/wiki/index.php?title=Formats:pepXML">
pep xml file format</a> for further reference. This file can be used as input to 
some of the tools in the 
<a href="http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP">
Transproteomic Pipeline</a>.
</li>

<li>
<b>percolator.log.txt</b>: a log file containing a copy of all messages that were
printed to stderr.
</li>
</ul>

<b>Options:</b>

<ul>

<li>
<code>--help </code> &ndash; Dispaly usage and options of percolator.</li>

<li>
<code>--protein &lt;T|F&gt;</code> &ndash;  Outoyt protein level probabilities</li>

<li>
<code> --xmloutput &lt;string&gt;</code> &ndash;  path to file in xml-output foramt (pout)</li>

<li>
<code>--stdinput &lt;T|F&gt;</code> &ndash;  Read xml-input format (pin) from standard</li>
<note>Maybe use &quot;-&quot; for input argument?</note>

<li>
<code>--decoy-xml-output &lt;T|F&gt;</code> &ndash;  Include decoys (PSMs,peptides and/or proteins) in the xml-ouput. Only available if <code>--xmloutput</code> is used.</li>

<li>
<code>--decoy-prefix &lt;string&gt;</code> &ndash; Option for single SQT file mode defining the 
name pattern used for shuffled data base. Typically set to random_seq.</li>

<li>
<code>--c-pos &lt;float&gt;</code> &ndash; Penalty for mistakes made on positive 
examples. Set by cross validation if not specified.</li>

<li>
<code>--c-neg &lt;float&gt;</code> &ndash; Penalty for mistake made on negative 
examples. Set by cross validation if not scpecified or <code>--c-pos</code> not specified.</li>
<li>
<code>--train-fdr &lt;value&gt;</code> &ndash; False discovery rate thereshold to define 
posetive examples in training. Set by cross validation if 0. Default is 0.01.</li>

<li>
<code>--test-fdr &lt;value&gt;</code> &ndash; False discovery rate threshold for evaluating best 
cross validation result and the reported end result. Default is 0.01. 
<br><note> Set to 1.0? Keep all PSMs for thresholding in downstream function</note></li>

<li>
<code>--maxiter &lt;int&gt;</code> &ndash; Maximum number of iterations. </li>

<li>
<code>  --train-ratio &lt;float&gt;</code> &ndash;  Fraction of the negative data set to be used 
as train set when on;y providing one negative set, remaining examples will be used as test set. 
Set to 0.6 defualt.</li>

<li>
<code>--output-feature-file &lt;string&gt;</code> &ndash; Output the computed features to the 
given file in tab-delimited format. A file with the features with the given file name will be 
created.</li>

<li>
<code>--input-feature-file &lt;filename&gt;</code> &ndash; Input files are given as a 
tab-delimited file. In this case the only argument should be a file name of the data file. 
The tab-delimited fields should be id &lt;tab&gt; label &lt;tab&gt; feature1 &lt;tab&gt; ... &lt;
tab&gt; featureN &lt;tab&gt; peptide &lt;tab&gt; proteinId1 &lt;tab&gt; ... &lt;tab&gt; 
proteinIdM Labels are interpreted as 1 -- positive set and test set, -1 -- negative</note> set. When the 
--doc option is ture, the first and second feature (third and fourth column) should contain the retention 
time and difference between observed and calculated mass.</li>

<li>
<code>--output-weights &lt;string&gt; </code>&ndash; Output final weights to the given file.</li>

<li>
<code>--input-weights &lt;string&gt;</code> &ndash; Read initial weights from the given file 
(one per line).</li>

<li>
<code>--defult-direction &lt;int&gt;</code> &ndash; The most informative feature given as feature 
number, can be negated to indicate that a lower value is better.</li>
<note>???</note>

<li>
<code>--verbosity &lt;int&gt;</code> &ndash; Set verbosity of output: 0=no processing info, 
5=all, default is 2.</li>

<li>
<code>--unitnorm &lt;T|F&gt;</code> &ndash; Use unit normalization [0-1] instead of standard 
deviation normalization</li>

<li>
<code>--alpha &lt;float&gt;</code> &ndash; Probability with which a present protein emits an 
associated peptide (to be used jointly with the --protein option). Set by grid search of not 
specified.</li>

<li>
<code>--beta &lt;float&gt;</code> &ndash; Probability of the creation of a peptide from noise 
(to be used jointly with the --protein option). Set by grid search of not specified.</li>

<li>
<code>--gamma &lt;float&gt;</code> &ndash; Prior probability of that a protein is present in 
the sample ( to be used with the --protein option). Set by grid search of not specified.</li>

<li>
<code>--test-each-iteration &lt;T|F&gt;</code> &ndash; Measure performance on test set each 
iteration.</li>


<li>
<code>--override &lt;T|F&gt;</code> &ndash;  Override error check and do not fall back on default 
score vector in case of suspect score vector.</li>

<li>
<code>--seed &lt;int&gt;</code> &ndash; Setting seed of the random number generator. Default value 
is 0.</li>

<li>
<code>--klammer &lt;T|F&gt;</code> &ndash;  Retention time features calculated as in Klammer et al
.</li>

<li> 
<code>--doc &lt;T|F&gt; </code> &ndash; Include description of correct features.</li>

<li>
<code><note>--results &lt;filename&gt;</note></code> &ndash; Output tab delimited results to a 
file instead of stdout.<note>This should be hardcoded to our tab-delimited results file.</note>
</li>

<li>
<code>--decoy-results &lt;filename&gt;</code> &ndash; Output tab delimited results for decoys 
into a file.</li><note>This should be hardcoded to our tab-delimited file</note>

<li>
<code>--unique-peptides &lt;T|F&gt;</code> &ndash; Do not remove redundant peptides, keep all 
PSMs and exclude Peptide Level Probability.</li>

<li>
<code>--no-schema-validation &lt;T|F&gt;</code> &ndash; Skip validation of input file against xml 
schema.</li>

<li>
<code>--clock &lt;T|F&gt;</code> &ndash; Check time performance.</li>

<li>
<code>--allow-protein-group &lt;T|F&gt;</code> &ndash; Treat ties as if it were one protein (Only 
valid if option --protein is active).</li>

<li>
<code>--protein--level-pi0 &lt;T|F&gt;</code> &ndash; Use pi_0 value when calculating empirical 
q-values (Only valid if option --protein is active).</li>

<li>
<code>--empirical-protein-q  &lt;T|F&gt;</code> &ndash; Output empirical q-values 
(from target-decoy analysis). Only valid if option --protein is active.</li>

<li>
<code>--group-proteins &lt;T|F&gt;</code> &ndash; Proteins with same probabilities will be 
grouped (Only valid if option --protein is active).</li>

<li>
<code>--no-prune-proteins &lt;T|F&gt;</code> &ndash; Peptides with low score will not be pruned 
before calculating protein probabilities(Only valid if option --protein is active).</li>

<li>
<code>--deepness &lt;0|1|2|3&gt;</code> &ndash; Setting deepness 0 or 1 or 2 or 3 from high 
deepness to low deepness(less computational time) of the grid search for Alpha,Beta and 
Gamma estimation(Only valid if option --protein is active). Default value is 3.</li>

<li>
<code>--exp-mass &lt;T|F&gt;</code> &ndash; Include the experimental mass in the output file.</li><note>I think this should 
always be true</note>


<!--
<li>
<code>--pi-zero &lt;value&gt;</code> &ndash; 
The estimated proportion of target scores that are drawn according to
the null distribution.  Default=1.0.</li>

<li>
<code>--mod-mass-format &lt;mod-only|total|separate&gt;<code> &ndash;
Specify how sequence modifications are reported in various ouptut
files.  Each modification is reported as a number enclosed in square
braces following the modified reside; however, the number may
correspond to one of three different masses: (1) 'mod-only' reports
the value of the mass shift induced by the modification; (2) 'total'
reports the mass of the residue with the modification (residue mass
plus modification mass); (3) 'separate' is the same as 'mod-only', but
multiple modifications to a single amino acid are reported as a
comma-separated list of values. For example, suppose amino
acid <i>D</i> has an unmodified mass of 115 as well as two
modifications of masses +14 and +2.  In this case, the amino acid
would be reported as D[16] with 'mod-only', D[131] with 'total', and
D[14,2] with 'separate'.</li>

<li>
<code>--fileroot &lt;string&gt;</code> &ndash; 
The <code>fileroot</code> string will be added as a prefix to all
output file names. Default = none.</li>

<li>
<code>--output-dir &lt;filename&gt;</code> &ndash; 
The name of the directory where output files will be created.
Default = crux-output.</li>

<li>
<code>--overwrite &lt;T|F&gt;</code> Replace existing files if true (T) or
fail when trying to overwrite a file if false (F). Default =  F.

<LI>
<code>--parameter-file &lt;filename&gt;</code> &ndash;
A file containing command-line or additional parameters.  See the <a
href=parameter-file.html>parameter documentation</a> page for details.</li>

<LI> 
<code>--feature-file &lt;string&gt;</code> &ndash;
Optional file in which to write the features. Default = none.</li>

<LI>
<code>--verbosity &lt;0-100&gt;</code> &ndash; Specify the verbosity of the
current processes. Each level prints the following messages, including
all those at lower verbosity levels: 0-fatal errors, 10-non-fatal
errors, 20-warnings, 30-information on the progress of execution,
40-more progress information, 50-debug info, 60-detailed debug info.
Default = 30.</li>
-->

</ul>


<hr></hr>

<a href="crux.html">Crux home</a>

</blockquote>
</body>
</html>
